src/main/java/codeu/controller/ParsingEngine.java

```diff
@@ -0,0 +1,33 @@
+package codeu.controller;
+
+import codeu.model.data.BotRequest;
+import codeu.model.data.BotStructure;
+import codeu.model.data.Message;
+import codeu.model.data.RoutingEngine;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+
+public class parsingEngine {
+ 	public void ParseCommands(Message message){
+ 		
```

Comment:
I think having an explanation of why the regex is constructed is useful. Explaining examples of what the regex matches and each parameter group explains what we are parsing.